pkg/descheduler/controllers/migration/arbitrator/handler.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package arbitrator
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/koordinator-sh/koordinator/apis/scheduling/v1alpha1"
+)
+
+// arbitrationHandler implement handler.EventHandler
+type arbitrationHandler struct {
+	handler.EnqueueRequestForObject
+	client     client.Client
+	arbitrator Arbitrator
+}
+
+func NewHandler(arbitrator Arbitrator, client client.Client) handler.EventHandler {
+	return &arbitrationHandler{
+		EnqueueRequestForObject: handler.EnqueueRequestForObject{},
+		arbitrator:              arbitrator,
+		client:                  client,
+	}
+}
+
+// Create call Arbitrator.Create
+func (h *arbitrationHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	if evt.Object == nil {
+		enqueueLog.Error(nil, "CreateEvent received with no metadata", "event", evt)
+		return
+	}
+	// get job
+	job := &v1alpha1.PodMigrationJob{}
+	err := h.client.Get(context.TODO(), types.NamespacedName{
+		Name:      evt.Object.GetName(),
+		Namespace: evt.Object.GetNamespace(),
+	}, job)
+	if err != nil {
+		// if err, add job to the workQueue directly.
+		enqueueLog.Error(nil, "Fail to get PodMigrationJob", "PodMigrationJob", types.NamespacedName{
+			Name:      evt.Object.GetName(),
+			Namespace: evt.Object.GetNamespace(),
+		})
+		q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+			Name:      evt.Object.GetName(),
+			Namespace: evt.Object.GetNamespace(),
+		}})
+		return
+	}
+	h.arbitrator.AddPodMigrationJob(job)
+}
+
+// Update implements EventHandler.
+func (h *arbitrationHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	switch {
+	case evt.ObjectNew != nil:
```

Comment:
Here you only need to deal with evt.ObjectNew.

Reply:
ok